cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreServiceUtils.java

```diff
@@ -172,21 +180,39 @@ public static HiveSupport checkHiveSupport(@Nullable ClassLoader hiveClassLoader
   }
 
   /**
-   * Rewrite the {@link HiveAuthFactory} class in the given source jar. The rewrite is for skipping kerberos
-   * authentication from the explore service container.
+   * Patch hive classes by bytecode rewriting in the given source jar. Currently it rewrite the following classes:
+   *
+   * <ul>
+   *   <li>
+   *     {@link HiveAuthFactory} - This is for skipping kerberos authentication from the explore service container.
+   *     {@link SessionState} - This is for workaround around a native memory leakage bug due to
```

Comment:
s/This is for workaround around a native memory leakage/This is to workaround a native memory leakage/